### PR TITLE
feat(staff): Create helper to check staff behind feature flag

### PR DESCRIPTION
**Pull request by schew2381**
I'm very open to different names for this helper.

Also simplified the permission check code for readability

### CODE REVIEW

**Review by nhsiehgit — COMMENTED** [review]
Can we get some permission tests?

otherwise lgtm

**Comment by schew2381** [comment]
> Can we get some permission tests?
> 
> otherwise lgtm

Done in https://github.com/getsentry/sentry/pull/63721/files#diff-d411a4700e3c3ac44a9d392f7226c8f3c45528006da3e2049e45e61fab5f3c57

File: tests/sentry/api/test_permissions.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        assert SuperuserPermission().has_permission(self.request, None)
+
+    def test_staff_permission(self):
+        self._activate_staff()
+        assert StaffPermission().has_permission(self.request, None)
+
+    def test_superuser_or_staff_feature_flagged_permission(self):
+        # Feature flag enabled
+        with self.feature("auth:enterprise-staff-cookie"):
+            assert not SuperuserOrStaffFeatureFlaggedPermission().has_permission(self.request, None)
```
Comment (cathteng) [inline tests/sentry/api/test_permissions.py:37]: should we also test that active superuser still returns false for this with the feature flag?

and likewise without the feature flag, active staff returns false

File: tests/sentry/api/test_permissions.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    def test_superuser_permission(self):
+        self._activate_superuser()
+        assert SuperuserPermission().has_permission(self.request, None)
+
+    def test_staff_permission(self):
+        self._activate_staff()
+        assert StaffPermission().has_permission(self.request, None)
+
+    def test_superuser_or_staff_feature_flagged_permission(self):
```
Comment (cathteng) [inline tests/sentry/api/test_permissions.py:37]: nit: you could split up this test so when you're checking with the feature flag, you can activate superuser first and check that it's False, then activate staff and check that it's True

and then have another test for without the feature flag and you activate staff first, then superuser. just an idea

**Review by cathteng — APPROVED** [review]

File: tests/sentry/api/test_permissions.py (hunk 2 above)
Comment (cathteng) [reply tests/sentry/api/test_permissions.py:37]: mostly so you don't have to deactivate superuser or staff

Comment (schew2381) [reply tests/sentry/api/test_permissions.py]: ok done

**Pull request merged**